lib/kube/proxy/forwarder.go

```diff
@@ -134,6 +134,8 @@ type ForwarderConfig struct {
 	DynamicLabels *labels.Dynamic
 	// LockWatcher is a lock watcher.
 	LockWatcher *services.LockWatcher
+
+	DisableImpersonationPermissionsCheck bool
```

Comment:
Instead of passing a boolean, why not pass the checker function/interface?

Also, missing godoc.

Reply:
Done